src/server/mod.rs

```diff
@@ -0,0 +1,313 @@
+
+                let mut max_dangling_height = 0;
+                let mut max_dangling_length = 0;
+                for dangling in &indexer_state.dangling_branches {
+                    if dangling.height() > max_dangling_height {
+                        max_dangling_height = dangling.height();
+                    }
+                    if dangling.len() > max_dangling_length {
+                        max_dangling_length = dangling.len();
+                    }
+                }
+                let db_stats_str = indexer_state
+                    .block_store
+                    .as_ref()
+                    .map(|db|
+                        db.database
+                        .property_value(rocksdb::properties::DBSTATS)
+                        .unwrap()
+                        .unwrap()
+                    );
+                let summary = Summary {
+                    block_count: indexer_state.block_count,
+                    date_time: indexer_state.date_time,
+                    uptime: indexer_state.time.clone().elapsed(),
+                    root_height: indexer_state.root_branch.height(),
+                    root_length: indexer_state.root_branch.len(),
+                    num_dangling: indexer_state.dangling_branches.len(),
+                    max_dangling_height,
+                    max_dangling_length,
+                    db_stats: db_stats_str.map(|s| DbStats::from_str(&s).unwrap()),
+                };
+
+                let mut leaves = indexer_state.root_branch.leaves.values().cloned();
+                let leaf = leaves.find(|leaf| &leaf.block.state_hash == best_chain.first().unwrap_or(&root_hash)).unwrap();
+                let ledger = leaf.get_ledger().clone();
+
+                tokio::spawn(async move {
+                    debug!("Handling connection");
+                    if let Err(e) = handle_conn(conn, block_store_readonly, best_chain, ledger, summary).await {
+                        error!("Error handling connection\n{e}");
+                    }
+                    debug!("Removing readonly instance at {}", secondary_path.display());
+                    tokio::fs::remove_dir_all(&secondary_path).await.ok();
+                });
+            }
+        }
+    }
+}
+
+#[instrument(level = "info")]
+async fn handle_conn(
+    conn: LocalSocketStream,
+    db: BlockStoreConn,
+    best_chain: Vec<BlockHash>,
+    ledger: Ledger,
+    summary: Summary,
+) -> Result<(), anyhow::Error> {
+    let (reader, mut writer) = conn.into_split();
+    let mut reader = BufReader::new(reader);
+    let mut buffer = Vec::with_capacity(128);
+    let _read = reader.read_until(0, &mut buffer).await?;
+
+    let mut buffers = buffer.split(|byte| *byte == 32);
+    let command = buffers.next().unwrap();
+
+    let command_string = String::from_utf8(command.to_vec())?;
+
+    match command_string.as_str() {
+        "account" => {
```

Comment:
add final `\0` here

Reply:
I got errors with `\0` at the end. I assumed it was because there was a pk before `\0`